Colony v0.2 - Adding Proxmox content (#64)
konstructio/colony-docs: docs/install/proxmox/proxmox-setup.md

```shell
ssh -i <private-key> kray@<ip-address>
```

### Install Docker

1. Install docker [for details refer to their documentation here.](https://github.com/docker/docker-install)

    ```shell
    curl -fsSL https://get.docker.com -o get-docker.sh
    sh get-docker.sh
    ```

2. Add your user to the `docker group` and restart your shell. You can exit out of your session and login right back

    ```shell
    usermod -aG docker kray
    ```

3. Run a test docker container

    ```shell
    docker run docker.io/hello-world
    ```

## Set up Colony

### Download Colony

Use the following command to download the Colony binary

```shell
wget https://objectstore.nyc1.civo.com/konstruct-assets/colony/v0.2.0-rc4/colony_Linux_x86_64.tar.gz
tar xvf colony_Linux_x86_64.tar.gz
sudo mv colony /usr/local/bin
```

### Install Colony
Install colony. Better wording would be creaste a Colony cluste or something. Check the interface you are listening on with `ip a` command.
Colony will use this interface to listen to DHCPOFFER traffic and respond with boot files.

``` shell

> colony init \
>     --apiKey $YOUR_COLONY_API_KEY \
>     --loadBalancerInterface eth0 \
>     --loadBalancerIP <loadbalancer>

Reviewer: ```suggestion
colony init \
    --api-key=$COLONY_API_KEY \
    --load-balancer-interface=$INTERFACE \
    --load-balancer-ip=$LOAD_BALANCER_IP
```

---
title: Installing Colony with Promox
description: Follow the steps to install Colony with Proxmox
sidebar_position: 2
---

## Installing Colony with Proxmox

> This guide walks through the setup of Laptop-0, a machine for running Colony, and 3 additional VM's for Colony to auto discovery and install a K3S cluster.  Some items to note:

Reviewer 2: End of the document we are referring to Talos
```suggestion
This guide walks through the setup of Laptop-0, a machine for running Colony, and 3 additional VM's for Colony to auto discovery and install a Talos cluster.  Some items to note:
```

- Colony requires Docker.

> - If you have a firewall on your host machine (laptop 0) it needs to allow the following ports (ports TBA).

Reviewer 2: This was a half thought out idea, let's remove it.

Author: Feel free to just suggest an edit for what you think should be removed.

> - If you have a firewall on your host machine (laptop 0) it needs to allow the following ports (ports TBA).

Reviewer 2: ```suggestion

```

Warning: Colony will be able to PXE boot all machines in your network.

## Getting Started

> The steps here are optional, but we recommend completing these to create a Proxmox template VM that you can clone to for better performance and asset creation.

Reviewer 2: ```suggestion
The steps here are optional, but we recommend completing these to create a Proxmox template VM that you can clone to quickly create virtual machines to experiment with. You can move on *add link here* if you have an existing virtual machine that you want to use as Laptop-0. 
```

Author: What link are you suggesting we add here?

### Debian image

> These steps use a `debian cloudinit image for laptop VM`.

Reviewer 2: ```suggestion
These steps use a `debian cloudinit image for laptop VM`. Following set of commands should be in your Proxmox host machine. You can either SSH into the host or use the shell in Proxmox UI. To access the shell click on `Datacenter` > your node (choose one if you have multiple > `shell`.
```

- You will need to install `libguestfs-tools` to interact with cloud-init image.

    ```shell
    apt update
    apt install libguestfs-tools
    ```

> - Commands should be run in Proxmox shell, which can be accessed through the Proxmox UI through `datacenter > <node-name> > shell`, or through ssh.

Reviewer 2: ```suggestion

```

> - Might to prepend sudo if you running user different than `root`

Reviewer 2: Maybe have this around a warning box? What do you think?

Reviewer 3: Seems also the sentence is incomplete? I also wrapped `sudo` since it's a command into a code box.

```suggestion
- Might need to prepend `sudo` if you running user different than `root`
```

> Download debian cloud init image [debian download image](https://cloud.debian.org/images/cloud/). Create the directory anywhere. Select the correct architecture. This example runs on x86

Reviewer 2: ```suggestion
Start by creating a directory. This will contain temporary artifacts we need to create template VM. Then download Debian cloud-init image [debian download image](https://cloud.debian.org/images/cloud/). 
```

```shell
mkdir debian-golden-img

wget https://cloud.debian.org/images/cloud/bookworm/20241004-1890/debian-12-genericcloud-amd64-20241004-1890.qcow2

```

### Guest agent and VIM

> Install `qemu-guest-agent` and `vim`. This enables information to report back from the guest machine to the host. Allows power off commands to pass to the guest.

Reviewer 2: ```suggestion
We will use `virt-customize` to install `qemu-guest-agent` and `vim`. The qemu-guest-agent is used to provide a communication channel between a virtual machine and its Proxmox. enabling features like shutdown, time synchronization, and better management of guest systems.
```

Reviewer 3: `qemu-guest-agent` should also be a code item.

```shell
virt-customize -a debian-12-genericcloud-amd64-20241004-1890.qcow2 --install qemu-guest-agent
virt-customize -a debian-12-genericcloud-amd64-20241004-1890.qcow2 --install vim
```

### Helm and Kubectl

> Install `kubectl` and `Helm` binary. Download and copy to `/usr/local/bin`

Reviewer 2: ```suggestion
Install Kubectl and Helm binaries and copy them over to the template VM to ensure these tools are available for managing Kubernetes clusters and Helm charts. Adding them to the template VM ensures that they are automatically included in any new VMs created from the template, eliminating the need for manual installation. 
```

For helm

```shell
wget https://get.helm.sh/helm-v3.16.2-linux-amd64.tar.gz
tar xvf helm-v3.16.2-linux-amd64.tar.gz
virt-customize -a debian-12-genericcloud-amd64-20241004-1890.qcow2 --copy-in ./linux-amd64/helm:/usr/local/bin/

rm -rf ./linux-amd64
```

For kubectl

```shell
# kubectl
wget "https://dl.k8s.io/release/v1.31.2/bin/linux/amd64/kubectl"
virt-customize -a debian-12-genericcloud-amd64-20241004-1890.qcow2 --copy-in ./kubectl:/usr/local/bin/
```

> ### Resize config

Reviewer 2: ```suggestion
### Finishing touches (? maybe)
```

Author: I think finishing touches is vague. Normally I'd use a heading that says what we're doing, if it's more than resizing something like... Final config steps

> Remove machine config and resize root disk to `32 Gb`. This allows new virtual machines created from the template to have unique machine ID.

Reviewer 2: Prepare virtual machine image by increasing the disk size to `32GB` and removing machine-specific information generated in previous step. This ensure that the image can be safely cloned or reused without conflicts.
